merlin/utils.py

```diff
@@ -559,33 +557,81 @@ def needs_merlin_expansion(
     return False
 
 
-def dict_deep_merge(dict_a, dict_b, path=None):
+def _both_inputs_are_dicts(dict_a: dict, dict_b: dict):
+    """
+    Helper function for `dict_deep_merge` to ensure both entries are dictionaries.
+
+    :param `dict_a`: A dict that we'll merge dict_b into
+    :param `dict_b`: A dict that we want to merge into dict_a
+    """
+    dict_a_is_dict = isinstance(dict_a, dict)
+    dict_b_is_dict = isinstance(dict_b, dict)
+    if not dict_a_is_dict or not dict_b_is_dict:
```

Comment:
All of this could be simplified, like msgs = [‘name is not a dict.’] for name in [A,B] if name not dict
error = ‘ ‘.join(msgs)

that syntax is wrong but you get the idea